larreco/QuadVtx/QuadVtx_module.cc

```diff
@@ -0,0 +1,522 @@
+// Chris Backhouse - c.backhouse@ucl.ac.uk - Oct 2019
+
+#include "larreco/QuadVtx/HeatMap.h"
+
+// C/C++ standard libraries
+#include <string>
+#include <iostream>
+
+// framework libraries
+#include "fhiclcpp/ParameterSet.h"
+#include "art/Framework/Core/ModuleMacros.h"
+#include "art/Framework/Core/EDProducer.h"
+#include "art/Framework/Principal/Event.h"
+#include "art/Framework/Principal/Handle.h"
+#include "art_root_io/TFileService.h"
+#include "art/Framework/Services/Registry/ServiceHandle.h"
+#include "canvas/Persistency/Common/Ptr.h"
+
+// LArSoft libraries
+#include "lardataobj/RecoBase/Hit.h"
+#include "lardataobj/RecoBase/Vertex.h"
+#include "lardata/DetectorInfoServices/DetectorPropertiesService.h"
+#include "larcore/Geometry/Geometry.h"
+
+#include "TGraph.h"
+#include "TH2.h"
+#include "TMatrixD.h"
+#include "TVectorD.h"
+
+// Private functions for this file
+namespace
+{
+  template<class T> inline T sqr(T x){return x*x;}
+  template<class T> inline T cube(T x){return x*x*x;}
+}
+
```

Comment:
There are various free functions in this file that should be placed in a namespace to keep them invisible to other parts of the program--e.g. `IntersectsCircle`, `LinesFromPoints`, `FindPeak3D`, etc.

Reply:
aren't they all within the 'quad' namespace?

Reply:
Oops, sorry I missed that.  In principle, even though it's in the `quad` namespace, if another library were to define the same function names/signatures in the `quad` namespace, that would be a violation of the one-definition rule.  The run-time is allowed to use the first definition it finds, which might not be what's desired.  To avoid that, it's better to put them in an anonymous namespace, which would make them visible only to this module.

Having said all that, it's pretty unlikely that someone would run into that problem.  It won't hold up the approval.